metaci/api/views/test_testmethod_perf.py

```diff
@@ -0,0 +1,435 @@
+    class Meta:
+        model = TestClass
+
+    repo = factory.SubFactory(RepositoryFactory)
+
+
+class TestMethodFactory(factory.django.DjangoModelFactory):
+    __test__ = False  # PyTest is confused by the classname
+
+    class Meta:
+        model = TestMethod
+
+    testclass = factory.SubFactory(TestClassFactory)
+    name = "GenericMethod"
+
+
+class TestResultFactory(factory.django.DjangoModelFactory):
+    __test__ = False  # PyTest is confused by the classname
+
+    class Meta:
+        model = TestResult
+
+    build_flow = factory.SubFactory(BuildFlowFactory)
+    method = factory.SubFactory(TestMethodFactory)
+    duration = 5
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+        django_get_or_create = ("username",)
+
+    email = factory.Sequence("user_{}@example.com".format)
+    username = factory.Sequence("user_{}@example.com".format)
+    password = factory.PostGenerationMethodCall("set_password", "foobar")
+    # socialaccount_set = factory.RelatedFactory(SocialAccountFactory, "user")
+
+
+class StaffSuperuserFactory(UserFactory):
+    is_staff = True
+    is_superuser = True
+
+
+class _TestingHelpers:
+    def debugmsg(self, *args):
+        print(*args)
+
+    @classmethod
+    def make_user_and_client(cls):
+        user = StaffSuperuserFactory()
+        client = APIClient()
+        client.force_authenticate(user)
+        response = client.get("/api/")
+        if response.status_code == 400 and "DisallowedHost" in str(response.content):
+            cls.debugmsg("**** YOU MAY NEED TO ADD AN ALLOWED_HOSTS TO YOUR TEST.PY")
+            raise (Exception(response))
+
+        assert response.status_code == 200, response.content
+        return client, user
+
+    def api_url(self, **kwargs):
+        params = urlencode(kwargs, True)
+        self.debugmsg("QueryParams", params)
+        return r"/api/testmethod_perf/?" + params
+
+    def find_by(self, fieldname, objs, value):
+        if type(objs) == dict:
+            objs = objs.get("results", objs)
+        return next((x for x in objs if x[fieldname] == value), None)
+
+    def api_call_helper(self, **kwargs):
+        self.debugmsg("Request", kwargs)
+        response = self.client.get(self.api_url(**kwargs, format="api"))
+        self.assertEqual(response.status_code, 200)
+        response = self.client.get(self.api_url(**kwargs))
+        self.assertEqual(response.status_code, 200)
+        objs = response.json()
+        self.debugmsg("Response", objs)
+        return objs["results"]
+
+    def stats_test_helper(self, stat):
+        return self.api_call_helper(include_fields=stat)
+
+    def identical_tests_helper(self, count, method_name="GenericMethod", **fields):
+        t1 = TestResultFactory(
+            build_flow__tests_total=1, method__name=method_name, **fields
+        )
+        for i in range(count - 1):
+            TestResultFactory(**fields, build_flow__tests_total=1, method=t1.method)
```

Comment:
This is definitely a matter of taste but I don't find `helper` as part of a name to be very, well, helpful for reading what the at a glance what the function does without needing to look at its code. I would probably call these `get_api_results` and `insert_identical_test_results`, and (even more subjective feedback) move the `stats_test_helper` one-liner inline.

Reply:
https://github.com/SFDO-Tooling/MetaCI/commit/89bd396f42daf6734c1369b1137ebdfdaba677f8